pkg/wireguard/wireguard.go

```diff
@@ -380,9 +380,9 @@ func (dev *Device) handlePeerEvent(logger *zap.Logger, peerEvents []PeerEvent) e
 				handleErr = handler.HandlePeerAdded(peerEvent)
 			}
 
-			peerEvents = slices.Delete(peerEvents, i, i+1)
-
 			if handleErr != nil {
+				peerEvents = slices.Delete(peerEvents, i, i+1)
+
```

Comment:
the code on line 391 is completely broken now as well... so this method is now a mess that needs to be fixed